streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -50,7 +50,7 @@
     // IQ may access this map.
     private Map<TaskId, Task> running = new ConcurrentHashMap<>();
     private Map<TopicPartition, Task> runningByPartition = new HashMap<>();
-    private Map<TopicPartition, Task> restoringByPartition = new HashMap<>();
+    private Map<TopicPartition, StreamTask> restoringByPartition = new HashMap<>();
```

Comment:
Some additional type improvements (if we agree to do this, we would need a follow up PR to do this in `2.0` and `trunk`, too)

Reply:
LGTM, as only active tasks can be restoring tasks. Are you suggesting to make this more strict to prevent potential bugs?

Reply:
Yes. IMHO, if we can improve type safety we should do it -- also, for newcomers it's easier to read the code.